includes/data-stores/class-wc-product-variable-data-store-cpt.php

```diff
@@ -286,84 +286,89 @@ public function read_price_data( &$product, $for_display = false ) {
 					'sale_price'    => array(),
 				);
 
-				$variation_ids = $product->get_visible_children();
+				// Allow 3rd parties to skip a full calculation of this data in favor of, for example, a background task.
+				$skip_price_data = apply_filters( 'woocommerce_variation_skip_price_data', false, $product );
 
-				if ( is_callable( '_prime_post_caches' ) ) {
-					_prime_post_caches( $variation_ids );
-				}
+				if ( ! $skip_price_data ) {
```

Comment:
can you move this condition as an `&&` clause inline 282? That way we don't have to indent the whole block. You might have to initialize `$transient_cached_prices_array[ $price_hash ]` though.

Reply:
I was able to simplify the branching. Please have another look when you have a chance! Thanks.